### ISSUE TITLE
output widget and matplotlib inline/widget

### ISSUE THREAD

**Issue by jdrylewicz** [jupyter-widgets/ipywidgets#2937]
Hi,
I am running the code below using %matplotlib inline and the output appears on a small window which cannot be expanded. My only option is to scroll - which is not what I want.

The alternative %matplotlib widget displays the image properly but when I use interact with that configuration, the plots does not refresh when I change an input widget.

Please let me know what I am doing wrong.

![Screen Shot 2020-08-02 at 9 33 23 PM](https://user-images.githubusercontent.com/62407823/89130751-ce3fda80-d507-11ea-91f8-8a7341d0c006.png)

`%matplotlib inline

out = widgets.Output(layout={'border': '1px solid black'})
with out:
    fig = plt.figure(figsize=(20,50))
    ax = fig.add_subplot(1,1,1)

     
# does not show the figure
display(out)
    
x = np.linspace(0, 10, 200)        
with out:
    ax.plot(x, x**2)
    show_inline_matplotlib_plots()`

versions:

jupyter core     : 4.6.3
jupyter-notebook : 5.7.5
qtconsole        : 4.3.1
ipython          : 7.16.1
ipykernel        : 5.3.4
jupyter client   : 6.1.6
jupyter lab      : 2.2.2
nbconvert        : 5.6.1
ipywidgets       : 7.5.1
nbformat         : 5.0.7
traitlets        : 4.3.3

**Comment by ianhi** [comment]
> I am running the code below using %matplotlib inline and the output appears on a small window which cannot be expanded.

I can replicate this behavior in jupyter notebook. I think this may be a bug with the output widget in the notebook, that output should probably be expandable. I'm not sure what the root cause is unfortunately. Though I can offer two workarounds for you:
1. Use jupyterlab (worked as expected for me there)
2. use `display(fig)` instead
```python
import ipywidgets as widgets
import matplotlib.pyplot as plt
import numpy as np

%matplotlib inline
out = widgets.Output(layout={'border': '1px solid black'})

fig = plt.figure(figsize=(20,50))
ax = fig.add_subplot(1,1,1)


x = np.linspace(0, 10, 200)
with out:
    ax.plot(x, x**2)
    widgets.interaction.show_inline_matplotlib_plots()
display(fig)
```
> The alternative %matplotlib widget displays the image properly but when I use interact with that configuration, the plots does not refresh when I change an input widget.

I'm surprised by this. Can you post an example of that? It may also be helpful to look at the final examples here: https://github.com/matplotlib/ipympl/blob/master/examples/ipympl.ipynb and in particular the usage of 
```python
    fig.canvas.draw()
    fig.canvas.flush_events()
```

(also quick tip: use three backticks to format code blocks on github see https://docs.github.com/en/github/writing-on-github/creating-and-highlighting-code-blocks)

**Comment by jdrylewicz** [comment]
yes - this plot shows fine with display - however below is an example where it does not. I use the notebook because I then convert the page using voila (I haven't tried to do so from jupyterlab yet).

In the example below switching to %matplotlib widget does not work great either.

```
%matplotlib inline

w_input = widgets.Select(
    options=['a', 'b'],
    value='a',
    description='metric:',
    rows=2,
    style={'description_width': 'initial'},
    disabled = False
)

def myplot(var):
    fig = plt.figure(1,figsize=(20,50))
    ax = fig.add_subplot(1,1,1)
    plt.show()
    return
    

output = interactive_output(myplot, {"var":w_input})  
display(w_input,output)
```
